tests/func/test_virtual_directory.py

```diff
@@ -182,7 +182,7 @@ def test_partial_checkout_and_update(M, tmp_dir, dvc, remote):
 
     assert dvc.pull("dir/subdir") == M.dict(
         added=[join("dir", "")],
-        fetched=1,
+        fetched=3,
```

Comment:
Is this an intentional change? Previously, we could fetch subset of files, but after this PR, we fetch everything.

I was not aware about this change, which I just noticed when someone asked in https://discuss.dvc.org/t/why-dvc-pulls-full-dataset-instead-of-a-single-file/1897.

This breaks an important scenario for virtual directory, see https://dvc.org/doc/user-guide/data-management/modifying-large-datasets#modifying-remote-datasets.

Do we have no means to fetch a subset, @efiop?